notebooks/Cross Validation Primer.ipynb

```diff
@@ -0,0 +1,432 @@
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Picking a Training Set"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "import numpy as np\n",
+    "import matplotlib.pyplot as plt\n",
+    "import pandas as pd\n",
+    "import sklearn as skl\n",
+    "import numpy as np\n",
+    "\n",
+    "import scripts.load_data as load\n",
+    "\n",
+    "import seaborn as sns\n",
+    "\n",
+    "%matplotlib inline"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from sklearn.linear_model import Ridge\n",
+    "from sklearn.ensemble import RandomForestRegressor\n",
+    "from sklearn.tree import DecisionTreeRegressor\n",
+    "from sklearn.preprocessing import PolynomialFeatures, StandardScaler\n",
+    "from sklearn.pipeline import make_pipeline\n",
+    "\n",
+    "#set the seed so we get the same data each time we run the cell\n",
+    "np.random.seed(0)\n",
+    "\n",
+    "#define a model to generate fake data\n",
+    "def data_model(x, sigma=0.1):\n",
+    "    base_model = np.log(x)\n",
+    "    noise = sigma * np.random.randn(len(x))\n",
+    "    return base_model + noise\n",
+    "\n",
+    "#Generate some fake data and keep a subset of them\n",
+    "#for training\n",
+    "#***Try different values for N***\n",
+    "N_split = 10\n",
+    "x_plot = np.linspace(1,100,100)\n",
+    "x_train = x_plot[:N_split]\n",
+    "y_train = data_model(x_train)\n",
+    "\n",
+    "#also make validation data\n",
+    "x_val = x_plot[N_split:]\n",
+    "y_val = data_model(x_val)\n",
+    "# create matrix versions of these arrays\n",
+    "X_train = x_train[:, np.newaxis]\n",
+    "X_val = x_val[:, np.newaxis]\n",
+    "X_plot = x_plot[:, np.newaxis]\n",
+    "\n",
+    "colors = ['teal', 'yellowgreen', 'gold', 'red']\n",
+    "lw = 0\n",
+    "plt.figure(figsize = (8,6))\n",
+    "plt.scatter(x_train, y_train, color='navy', s=30, marker='o', label=\"training points\", linewidth=lw)\n",
+    "plt.xlabel('x')\n",
+    "plt.ylabel('y')\n",
+    "\n",
+    "plt.scatter(x_val, y_val, marker='o', color='cornflowerblue', alpha=0.5,linewidth=lw, label=\"validation data\")\n",
+    "\n",
+    "plt.ylim([0,5])\n",
+    "lw =2\n",
+    "for count, degree in enumerate([2, 4, 5]):\n",
+    "    model = make_pipeline(PolynomialFeatures(degree), Ridge())\n",
+    "    #fit to training data\n",
+    "    model.fit(X_train, y_train)\n",
+    "    #apply to validation data\n",
+    "    y_predicted = model.predict(X_val)\n",
+    "    #apply again, just to show plot\n",
+    "    y_plot = model.predict(X_plot)\n",
+    "    plt.plot(x_plot, y_plot, color=colors[count], linewidth=lw, alpha=0.5,\n",
+    "             label=\"degree %d\" %(degree))\n",
+    "\n",
+    "plt.legend(loc='lower right')\n"
```

Comment:
Add a `;` to the end of this line to get rid of the output
`<matplotlib.legend.Legend at 0x110391ac8>`